src/back/glsl/mod.rs

```diff
@@ -844,6 +878,31 @@ impl<'a, W: Write> Writer<'a, W> {
             writeln!(self.out, ";")?
         }
 
+        // Bake expressions, this prevents exponential explosions
+        for expr in func
+            .expressions
```

Comment:
I don't think we can go through all function expressions and create local variables for them in one place. For example, if an expression is introduced within the loop (related to #452), the relevant temporary variable needs to be within the loop.

Reply:
I've been thinking about this and working around this now seems to be a waste of time since the issue is so high priority and we would have to change this later, I think it's better to wait for us to solve the ownership issue and return to this later

Reply:
This code is not going to be correct in most programs though. I.e. Anything that's in the loop can't be baked here. So it doesn't appear to be reasonable to try landing this as a temporary solution even. Maybe it's not that high a priority? Currently the problem is that some IRs may cause the backend to explode, but at least it's correct all the time.

Reply:
I'm not saying to land this, I'm saying we should close this PR, figure out the owner ship problem and then fix the backends

Reply:
yep, sounds reasonable!
I'm going to start working on the ownership rewrite tonight.